src/compute-client/src/protocol/history.rs

```diff
@@ -245,8 +252,8 @@ where
             self.commands.push(ComputeCommand::InitializationComplete);
         }
 
-        if allow_writes {
-            self.commands.push(ComputeCommand::AllowWrites);
+        for id in allow_writes {
+            self.commands.push(ComputeCommand::AllowWrites(id));
```

Comment:
Should we pull this before `InitializationComplete`? I think it doesn't matter much as we don't do anything special for these commands during reconciliation, but it feels more obvious to have `InitializationComplete` as the last command.

(Previously this wasn't possible because `AllowWrites` wasn't allowed in the initialization phase, as it was the command that marked the transition from read-only to read-write computation phase. But I'm proposing to change that above.)